### PR TITLE
MM-52902 : Pasting text or image in center message input box shifts input box up unexpectedly

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52902

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx [hunk 1]
```diff
@@ -1040,7 +1040,7 @@ class AdvancedCreatePost extends React.PureComponent<Props, State> {
 
     handleUploadError = (err: string | ServerError, clientId?: string, channelId?: string) => {
         let serverError = err;
-        if (typeof serverError === 'string') {
+        if (typeof serverError === 'string' && serverError.length !== 0) {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/advanced_create_post/advanced_create_post.tsx:1043]: only change is this one.
Rest everything else if converting to functional component for no reason whatsoever :D

**Comment by amyblais** [comment]
@M-ZubairAhmed There are some failing checks here.

**Review by larkox — APPROVED** [review]

File: webapp/channels/src/components/message_submit_error.tsx [hunk 2]
```diff
@@ -1,68 +1,60 @@
...
+                            id='message_submit_error.sendAsMessageLink'
+                            defaultMessage='Click here to send as a message.'
+                        />
+                    </a>
                 </label>
             </div>
         );
     }
+
+    if (props.error?.message?.trim()?.length === 0) {
```
Comment (affkjuhyt) [inline webapp/channels/src/components/message_submit_error.tsx:49]: what happen when `props.error?` is null or undefined

**Review by harshilsharma63 — APPROVED** [review]
LGTM 👍 other than the failing CI step

File: webapp/channels/src/components/advanced_create_comment/advanced_create_comment.tsx [hunk 3]
```diff
@@ -1116,8 +1116,8 @@ class AdvancedCreateComment extends React.PureComponent<Props, State> {
         }
     };
 
-    handleUploadError = (err: string | ServerError | null, clientId: string | number = -1, _?: string, rootId = '') => {
-        if (clientId !== -1) {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/advanced_create_comment/advanced_create_comment.tsx:1120]: clientId can never be a number

File: webapp/channels/src/components/message_submit_error.tsx (hunk 2 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/message_submit_error.tsx:49]: error is never null, this component is only rendered when error is not null

**Review by yasserfaraazkhan — COMMENTED** [review]
Tested in center, RHS thread text boxes
there was no shift on text box observed when pasting text content.

**Review by yasserfaraazkhan — COMMENTED** [review]
@M-ZubairAhmed 
few (3) e2e tests related to slash command have failed, it is due to web server and not due to changes in this PR.

can you help in confirming if those are unrelated failures ?

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed 
> 
> few (3) e2e tests related to slash command have failed, it is due to web server and not due to changes in this PR.
> 
> 
> 
> can you help in confirming if those are unrelated failures ?

Can you please help run the e2e again just so we rule out flakiness

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan i verified that was an actual bug, i fixed it in my latest commit. Lets re run the e2e when the CI is green

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan all tests passed 100%

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**